addons/mrp_subcontracting_dropshipping/models/stock_warehouse.py

```diff
@@ -15,6 +15,64 @@ class StockWarehouse(models.Model):
+            res.update_global_route_dropship_subcontractor()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        # if all warehouses have resupply disabled, disable global route, until its enabled on a warehouse
+        if 'subcontracting_dropshipping_to_resupply' in vals or 'active' in vals:
+            if 'subcontracting_dropshipping_to_resupply' in vals:
+                # ignore when warehouse archived since it will auto-archive all of its rules
+                self._update_dropship_subcontract_rules()
+            self.update_global_route_dropship_subcontractor()
+        return res
+
+    def _update_dropship_subcontract_rules(self):
+        '''update (archive/unarchive) any warehouse subcontracting location dropship rules'''
+        subcontracting_locations = self._get_subcontracting_locations()
+        route_id = self._find_global_route('mrp_subcontracting_dropshipping.route_subcontracting_dropshipping',
+                                           _('Dropship Subcontractor on Order'))
+        warehouses_dropship = self.filtered(lambda w: w.subcontracting_dropshipping_to_resupply and w.active)
+        if warehouses_dropship:
+            self.env['stock.rule'].with_context(active_test=False).search([
+                ('route_id', '=', route_id.id),
+                ('action', '=', 'pull'),
+                ('warehouse_id', 'in', warehouses_dropship.ids),
+                ('location_src_id', 'in', subcontracting_locations.ids)]).action_unarchive()
+
+        warehouses_no_dropship = self - warehouses_dropship
+        if warehouses_no_dropship:
+            self.env['stock.rule'].search([
+                ('route_id', '=', route_id.id),
+                ('action', '=', 'pull'),
+                ('warehouse_id', 'in', warehouses_no_dropship.ids),
+                ('location_src_id', 'in', subcontracting_locations.ids)]).action_archive()
+
+    def update_global_route_dropship_subcontractor(self):
+        route_id = self._find_global_route('mrp_subcontracting_dropshipping.route_subcontracting_dropshipping',
+                                           _('Dropship Subcontractor on Order'))
+        # if route has no pull rules, it means all warehouses have Dropship Subcontractor disabled
+        # Pick type is per company so we need to check rules per company to archive it, however
+        # the route is global so we need to check all rules regardless of company
+        all_rules = route_id.sudo().rule_ids.filtered(lambda r: r.active)
+        for company in self.company_id:
+            company_rules = all_rules.filtered(lambda r: r.company_id == company)
+            if not company_rules.filtered(lambda r: r.action == 'pull'):
+                company.dropship_subcontractor_pick_type_id.active = False
+            else:
+                company.dropship_subcontractor_pick_type_id.active = True
+
```

Comment:
```
  if not company_rules.filtered(lambda r: r.action == 'pull'):
      company.dropship_subcontractor_pick_type_id.active = False
  else:
      company.dropship_subcontractor_pick_type_id.active = True
```
could be written
`company.dropship_subcontractor_pick_type_id.active = not company_rules.filtered(lambda r: r.action == 'pull')`
same for route_id.active